core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DatalakeV4Controller.java

```diff
@@ -28,11 +31,46 @@ public class DatalakeV4Controller implements DatalakeV4Endpoint {
     @Inject
     private WorkspaceService workspaceService;
 
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
     @Override
+    @InternalOnly
     public StackViewV4Responses list(@Deprecated String environment, @TenantAwareParam String environmentCrn) {
         if (StringUtils.isBlank(environmentCrn) || StringUtils.isNotBlank(environment)) {
             throw new BadRequestException("environment param is deprecated, only environmentCrn should be filled.");
         }
         return stackOperations.listByEnvironmentCrn(workspaceService.getForCurrentUser().getId(), environmentCrn, List.of(StackType.DATALAKE));
     }
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        if (StringUtils.isBlank(platform)) {
+            throw new BadRequestException("Platform param is null or empty.");
+        }
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<VolumeSetAttributes.Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        String cloudPlatform = sdxDiskUpdateRequest.getCloudPlatform();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, cloudPlatform);
+    }
+
+    @Override
+    public void stopCluster(long stackId) throws Exception {
+        diskUpdateService.stopCluster(stackId);
+    }
+
+    @Override
+    public void startCluster(long stackId) throws Exception {
+        diskUpdateService.startCluster(stackId);
+    }
+
+    @Override
+    public void resizeDisks(long stackId, String instanceGroup) throws Exception {
+        diskUpdateService.resizeDisks(stackId, instanceGroup);
```

Comment:
you should put this into flow

Reply:
@horadla23 no it will be reused in case of datahubs. that is the reason i wrote a generic service which is being called here

Reply:
even if you are reusing it for DHs, for datalake you should put this into flow and check flow in datalake service, otherwise you will consume http thread for long and it will cause performance issue, all other features using the same logic, so pls do not create an exception here, but please use the same methodology and use flow for datalake operation

Reply:
generally saying, long lasting service logic should not be used in controller without flow

Reply:
since you are contacting nodes here with salt, it is recognized as long lasting operation

Reply:
done